spacy/sklearn.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
```

Comment:
I'm thinking we can have something like:

```python

try:
    from sklearn.base import BaseEstimator, TransformerMixin
except ImportError:
    BaseEstimator = object
    TransformerMixin = object
```

This way we can avoid the `scikit-learn` requirement.